File: LeetCode/Java/src/main/java/org/redquark/tutorials/leetcode/PalindromeNumber.java
package org.redquark.tutorials.leetcode;

/**
 * @author Anirudh Sharma
 */
public class PalindromeNumber {

    private static boolean isPalindrome(int x) {
        // Base condition
        if (x < 0) {
            return false;
        }
        // Store the number in a variable
        int number = x;
        // This will store the reverse of the number
        int reverse = 0;
        while (number > 0) {
            reverse = reverse * 10 + number % 10;
            number /= 10;
        }
        return x == reverse;
    }

    public static void main(String[] args) {
        System.out.println(isPalindrome(121));
        System.out.println(isPalindrome(-121));
        System.out.println(isPalindrome(10));
        System.out.println(isPalindrome(-101));
    }
}

File: LeetCode/Kotlin/src/main/kotlin/org/redquark/tutorials/leetcode/PalindromeNumber.kt
package org.redquark.tutorials.leetcode

fun isPalindrome(x: Int): Boolean {
    // Base condition
    if (x < 0) {
        return false
    }
    // Store the number in a variable
    var number = x
    // This will store the reverse of the number
    var reverse = 0
    while (number > 0) {
        reverse = reverse * 10 + number % 10
        number /= 10
    }
    return x == reverse
}

fun main() {
    println(isPalindrome(121))
    println(isPalindrome(-121))
    println(isPalindrome(10))
    println(isPalindrome(-101))
}